.chloggen/component-id-length-limit.yaml

```
@@ -0,0 +1,25 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: component
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Changes component.Type validation regex to only allow a max of 63 characters in a type name.
```

The fact that we use a regex is an implementation detail. I would phrase it in a way that is more meaningful to Go API end-users:
```suggestion
note: Restricts maximum length for `component.Type` to 63 characters.
```